doc/osi.sgml

```diff
@@ -34,42 +35,63 @@ information.
 Currently the target "osic1p" is implemented. This works for the Ohio Scientific
 Challenger 1P machine.
 
-<sect>Binary format<p>
+<sect>Program file formats<p>
 
-The standard binary output format generated by the linker for the osic1p target
-is a machine language program.<p>
+<descrip>
+  <tag/Binary, then text/
+  The standard binary output format generated by the linker for the osic1p
+  target is a pure machine language program.
+
+  For uploading into a real machine over its serial port or into an emulator,
+  that program must be converted into a text file that can be understood by
+  the 65V PROM monitor. For that purpose, the <bf/srec_cat/ program from <url
+  url="http://srecord.sourceforge.net/" name="the SRecord tool collection">
+  can be used.
+
+  Care must be taken that the <tt/-offset/ and <tt/-execution-start-address/
+  options for the <bf/srec_cat/ program correspond to the start address
+  of the executable.
+
+  Example for converting an executable "hello" file that was built for the
+  default start address &dollar;0200 to an uploadable file "hello.c1p":
 
-For uploading into a real machine over the serial port or into an emulator a
-program must be converted to a text file that is understood by the 65V PROM
-monitor. For this purpose the srec_cat program from the
-<url url="http://srecord.sourceforge.net/" name="SRecord">
-tool collection can be used.
+  <tscreen><verb>
+  srec_cat hello -bin -of 0x200 -o hello.c1p -os -esa=0x200
+  </verb></tscreen>
 
-Care must be taken that the -offset and -execution-start-address options for the srec_cat
-program correspond to the start address of the executable.
+  <tag/Hybrid/
+  The linker can create an alternate format that contains two parts:
+  <enum>
+    <item>A text header that is understood by the 65V PROM monitor.
+          It is a boot loader that reads the second part.
+    <item>The default binary code that is described above.
+  </enum>
 
-Example for converting an executable "hello" that was built for the default
-start address &dollar;0200 to an uploadable file "hello.c1p":
+  You can make the alternate format by adding the option <tt/-u __BOOT__/ to
+  <tt/cl65/'s or <tt/ld65/'s command lines.
 
-<tscreen>
-srec_cat hello -binary -offset 0x200 -o hello.c1p -Ohio_Scientific -execution-start-address=0x200
-</tscreen>
+  This format doesn't need to be converted. It is smaller than the text-only
+  format.  But, it cannot be loaded by <url
+  url="http://www.pcjs.org/docs/c1pjs/" name="C1Pjs">; you must use the
+  SRecord-produced text-only format with that emulator.
```

Comment:
In the intro document, I describe how to load a program into a real, physical machine and two emulators.  The real box and one of the emulators can handle the new format.  The Javascript-based emulator can't handle it because that program doesn't emulate the cassette player device.  Instead, it auto-types text-only files through the keyboard emulation.